python/hsfs/core/hudi_engine.py

```diff
@@ -235,6 +235,20 @@ def _setup_hudi_read_opts(self, hudi_fg_alias, read_options):
 
         return hudi_options
 
+    def reconcile_hudi_schema(self, spark_engine, hudi_fg_alias, read_options):
```

Comment:
I am wondering if we should use the same logic in the `append_feature`. E.g. take the schema from the hive table instead of the rather hacky `spark.get_empty_appended_dataframe()` method. So we are consistent.

Reply:
makes sense! I changed it for the java engine as well.